File: LeetCode/Easy/Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
    let numberHash: Map<number, number> = new Map()
    for (let i = 0; i < nums.length; i++) {
        let curr = nums[i]
        let compliment:number = target - curr;

        if (numberHash.get(compliment) !== undefined) {
            return [numberHash.get(compliment),i]
        }
        numberHash.set(curr, i);
    }
    return null;
};

File: LeetCode/Python/Easy/Two_Sum.py
class Solution:
    def twoSum(self, nums, target):
        dict = {}
        for x in range(len(nums)):
            compliment = target - nums[x]
            if (compliment in dict):
                return [dict.get(compliment), x]
            dict[nums[x]] = x


solution = Solution
print(solution.twoSum(solution, [3,2,4], 6))